crates/pallet-bridges-initializer-currency-swap/src/mock/v1.rs

```diff
@@ -0,0 +1,126 @@
+pub(crate) const EXISTENTIAL_DEPOSIT_EVM: u64 = 20;
+
+type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Test>;
+type Block = frame_system::mocking::MockBlock<Test>;
+
+pub(crate) type AccountId = u64;
+pub(crate) type EvmAccountId = u64;
+type Balance = u64;
+
+frame_support::construct_runtime!(
+    pub enum Test where
+        Block = Block,
+        NodeBlock = Block,
+        UncheckedExtrinsic = UncheckedExtrinsic,
+    {
+        System: frame_system,
+        Balances: pallet_balances::<Instance1>,
+        EvmBalances: pallet_balances::<Instance2>,
+    }
+);
+
+impl frame_system::Config for Test {
+    type BaseCallFilter = frame_support::traits::Everything;
+    type BlockWeights = ();
+    type BlockLength = ();
+    type DbWeight = ();
+    type RuntimeOrigin = RuntimeOrigin;
+    type RuntimeCall = RuntimeCall;
+    type Index = u64;
+    type BlockNumber = u64;
+    type Hash = H256;
+    type Hashing = BlakeTwo256;
+    type AccountId = AccountId;
+    type Lookup = IdentityLookup<AccountId>;
+    type Header = Header;
+    type RuntimeEvent = RuntimeEvent;
+    type BlockHashCount = ConstU64<250>;
+    type Version = ();
+    type PalletInfo = PalletInfo;
+    type AccountData = pallet_balances::AccountData<Balance>;
+    type OnNewAccount = ();
+    type OnKilledAccount = ();
+    type SystemWeightInfo = ();
+    type SS58Prefix = ();
+    type OnSetCode = ();
+    type MaxConsumers = ConstU32<16>;
+}
+
+type BalancesInstanceNative = pallet_balances::Instance1;
+type BalancesInstanceEvm = pallet_balances::Instance2;
+
+impl pallet_balances::Config<BalancesInstanceNative> for Test {
+    type Balance = u64;
+    type RuntimeEvent = RuntimeEvent;
+    type DustRemoval = ();
+    type ExistentialDeposit = ConstU64<EXISTENTIAL_DEPOSIT_NATIVE>;
+    type AccountStore = System;
+    type MaxLocks = ();
+    type MaxReserves = ();
+    type ReserveIdentifier = [u8; 8];
+    type WeightInfo = ();
+}
+
+impl pallet_balances::Config<BalancesInstanceEvm> for Test {
+    type Balance = u64;
+    type RuntimeEvent = RuntimeEvent;
+    type DustRemoval = ();
+    type ExistentialDeposit = ConstU64<EXISTENTIAL_DEPOSIT_EVM>;
+    type AccountStore = StorageMapShim<
+        pallet_balances::Account<Test, BalancesInstanceEvm>,
+        frame_system::Provider<Test>,
+        EvmAccountId,
+        pallet_balances::AccountData<Balance>,
+    >;
+    type MaxLocks = ();
+    type MaxReserves = ();
+    type ReserveIdentifier = [u8; 8];
+    type WeightInfo = ();
+}
+
+parameter_types! {
+    pub NativeTreasury: AccountId = 4200;
+}
+
+// This function basically just builds a genesis storage key/value store according to
+// our desired mockup.
+pub fn new_test_ext_with(genesis_config: GenesisConfig) -> sp_io::TestExternalities {
+    let storage = genesis_config.build_storage().unwrap();
+    storage.into()
+}
+
+pub fn runtime_lock() -> std::sync::MutexGuard<'static, ()> {
```

Comment:
This runtime lock should probably be shared across the versions - as the scope of it is the access to a global state, not a particular version's state.

Reply:
https://github.com/humanode-network/humanode/pull/716/commits/55ad986186808aad0dc49816ce9b07638f84329e